### DISCUSSION TITLE
ToolAssistantToolsFunction and Tool no longer found

### DISCUSSION THREAD

**Discussion by orome** [camel-ai/camel#483] [General]
What should I now be doing instead of 
```python
from openai.types.beta.threads.run_create_params import ToolAssistantToolsFunction, Tool
```

**Comment by Wendong-Fan** [comment]
Hey @orome , thanks for the issue,

this has been solved in https://github.com/camel-ai/camel/pull/475
please pull the update repo, thanks!

**Reply by orome** [reply]
What's the replacement though? What is now the type of what was formerly `ToolsAssistantToolsFunction` (and for that matter `Tool`) and where is it defined?

**Reply by Wendong-Fan** [reply]
Hey @orome , we just simply removed `ToolsAssistantToolsFunction` since this it was just imported to do a check, we think this check is not necessary.  
`ToolsAssistantToolsFunction` is from openai, from their answer it has been renamed to `FunctionToolCall`
FYR: https://github.com/openai/openai-python/issues/1242
